pkg/security/x509source.go

```diff
@@ -312,6 +323,115 @@ func (x *x509source) requestFromSentry(ctx context.Context, csrDER []byte) ([]*x
 	return workloadcert, nil
 }
 
+// maybeWriteSVIDToDir writes the current SVID to the configured directory if
+// configured. Does this atomically using unix symlinks.
+func (x *x509source) maybeWriteSVIDToDir() error {
+	if x.writeToDiskDir == nil || x.currentSVID == nil || x.trustAnchors == nil {
+		return nil
+	}
+
+	certs, pk, err := x.currentSVID.Marshal()
+	if err != nil {
+		return err
+	}
+
+	ta, err := x.trustAnchors.Marshal()
+	if err != nil {
+		return err
+	}
+
+	if err := atomicWrite(x.clock, *x.writeToDiskDir, map[string][]byte{
+		"tls.crt": certs,
+		"tls.key": pk,
+		"ca.crt":  ta,
+	}); err != nil {
+		return fmt.Errorf("unable to write SVID to disk: %w", err)
+	}
+
+	log.Infof("Wrote SVID to %q", *x.writeToDiskDir)
```

Comment:
To avoid ugly stuff on Windows

```suggestion
	log.Infof("Wrote SVID to '%s'", *x.writeToDiskDir)
```